src/libcore/intrinsics.rs

```diff
@@ -1335,6 +1337,29 @@ mod real_intrinsics {
   }
 }
 
+/// Checks whether `ptr` is properly aligned with respect to
+/// `align_of::<T>()`.
+fn is_aligned<T>(ptr: *const T) -> bool {
+    return ptr as usize % mem::align_of::<T>() == 0;
+}
+
+/// Checks whether the regions of memory starting at `src` and `dst` of size
+/// `count * size_of::<T>()` overlap.
+fn overlaps<T>(src: *const T, dst: *const T, count: usize) -> bool {
+    use crate::cmp::Ordering;
+    let src_usize = src as usize;
+    let dst_usize = dst as usize;
+    let size = mem::size_of::<T>() * count;
+    match src_usize.cmp(&dst_usize) {
+        // src < dst < src + offset
+        Ordering::Less => src_usize + size > dst_usize,
+        // dst < src < dst + offset
+        Ordering::Greater => dst_usize + size > src_usize,
+        // src == dst
+        Ordering::Equal => count != 0,
```

Comment:
```suggestion
        Ordering::Equal => size != 0,
```

To cover the `size_of::<T>() == 0` case as well.